src/pydocstyle/config.py

```diff
@@ -208,7 +208,10 @@ def _get_config(self, node):
         if path in self._cache:
             return self._cache[path]
 
-        config_file = self._get_config_file_in_folder(path)
+        if self._options.config_file:
```

Comment:
@jacebrowning , I just noticed a problem - in line 232 `self._get_config` is called recursively on the parent file in order to inherit its options, but the call is irrelevant because you'd just override the file path and use the same file from the cli (line 212).

Considering my previous comment, i think that ignoring the configuration hierarchy is definitely not the way to go since one config file is only defining a part of the final configuration.  In that case, the difference between specifying a path to a directory vs a path to a config file is little. I'd support both by calling `self._get_config_file_in_folder(self._options.config_path)`.